### ISSUE TITLE
NestedParenthesizedCalls and MethodCallWithArgsParentheses autocorrect produces code with syntax error

### ISSUE THREAD

**Issue by jonas054** [rubocop-hq/rubocop#9335]
With the following code
```yaml
# .rubocop.yml
AllCops:
  NewCops: enable

Style/FrozenStringLiteralComment:
  Enabled: false

Style/MethodCallWithArgsParentheses:
  Enabled: true
```

```ruby
# file.rb
a(b 1)
```

running `rubocop -a`

## Expected behavior

```ruby
# file.rb
a(b(1))
```

## Actual behavior

```console
file.rb:1:3: C: [Corrected] Style/MethodCallWithArgsParentheses: Use parentheses for method calls with arguments.
a(b 1)
  ^^^
file.rb:1:3: C: [Corrected] Style/NestedParenthesizedCalls: Add parentheses to nested method call b 1.
a(b 1)
  ^^^
file.rb:1:8: E: Lint/Syntax: unexpected token tRPAREN
(Using Ruby 2.4 parser; configure using TargetRubyVersion parameter, under AllCops)
a(b(1)))
       ^
```

```ruby
# file.rb
a(b(1)))
```

## Steps to reproduce the problem

As above.

## RuboCop version

```
$ rubocop -V
1.7.0 (using Parser 3.0.0.0, rubocop-ast 1.3.0, running on ruby 3.0.0 x86_64-linux)
```

I'm working on a fix for this myself. I think the easiest solution is to make `Style/NestedParenthesizedCalls` do nothing if `Style/MethodCallWithArgsParentheses` is enabled (which it isn't by default).
